lib/Album/AlbumMapper.php

```diff
@@ -273,6 +273,32 @@ public function removeFile(int $albumId, int $fileId): void {
 		$query->executeStatement();
 	}
 
+	public function removeFileWithOwner(int $fileId, string $ownerId): void {
+		// Get concerned albums.
+		$query = $this->connection->getQueryBuilder();
+		$rows = $query->select('album_id')
+			->from("photos_albums_files")
+			->where($query->expr()->eq("owner_id", $query->createNamedParameter($ownerId)))
+			->andWhere($query->expr()->eq("file_id", $query->createNamedParameter($fileId, IQueryBuilder::PARAM_INT)));
+		$query->executeStatement();
+
+		// Remove any occurrence of fileId when owner is ownerId.
+		$query = $this->connection->getQueryBuilder();
+		$query->delete("photos_albums_files")
+			->where($query->expr()->eq("owner_id", $query->createNamedParameter($ownerId)))
+			->andWhere($query->expr()->eq("file_id", $query->createNamedParameter($fileId, IQueryBuilder::PARAM_INT)));
+		$query->executeStatement();
```

Comment:
Is there a way to get some data from a delete call ? Ideally I would get album ids from the delete call, but not sure if that is possible.